### ISSUE TITLE
Editing config file from lazygit requires a restart for changes to take effect

### ISSUE THREAD

**Issue by peppy** [jesseduffield/lazygit#1158]
**Describe the bug**
Lazygit allows you to edit its configuration by pressing the 'e' hotkey from pane 1. After editing, my expectation would be that the new configuration is loaded, but it does not seem to be.

**To Reproduce**
1. Go to pane 1
2. Press 'e'
3. Edit the configuration (add a new custom command)
4. Save and return to lazygit

**Expected behavior**
The new command will work immediately, without a restart.

**Desktop (please complete the following information):**
 - OS: macOS (latest non-beta)
 - Lazygit Version 0.24.2

**Comment by jesseduffield** [comment]
Makes sense, I suspect that wouldn't be too hard to implement. I'll see if I find some time soon to work on this but in the meantime I'll chuck a 'good-first-issue' label on it

**Comment by jesseduffield** [comment]
@peppy are you able to give https://github.com/jesseduffield/lazygit/pull/1170 a go?

**Comment by peppy** [comment]
works as expected 💯

**Comment by c02y** [comment]
I'm sorry, how can I make it work?

And what is the meaning of "returning to gui", there is only tui for lazygit.

**Comment by peppy** [comment]
At some point I believe this did regress. I noticed that changes don't seem to stick anymore.

**Comment by c02y** [comment]
You may need to reopen it.

**Comment by joshgoebel** [comment]
In testing right now (on `main`) changing `commitAuthorShortLength` was instant (as soon as I switched based to lazygit).

**Comment by stefanhaller** [comment]
This was implemented in #3787.
